### PR TITLE
System Admin: add new Custom Field types

### PR DESCRIPTION
**Pull request by SKuipers**
Requires #1260.

Third piece of the Custom Fields refactoring.

- Adds new field types: Rich Text, Checkboxes, Radio, Time, File, Image, Yes/No, Number, Colour
  Screenshot 2021-03-01 at 2 53 23 PM

- Improves the layout and usability of the Add/Edit Custom Fields pages
- Enables optgroups in Dropdown fields using [] around an option name
  Screenshot 2021-03-01 at 5 08 25 PM

- Adds drag-drop ordering to the Custom Fields page
  Screenshot 2021-03-01 at 2 53 10 PM

**How Has This Been Tested?**
Locally.

**Screenshots**
Example of a variety of custom fields displayed through a Details table:
Screenshot 2021-03-01 at 5 06 03 PM

### CODE REVIEW

**Comment by SKuipers** [comment]
Thanks for the merge! I've rebased this one onto v22.0.00 (to make the changeset easier to read)

File: modules/System Admin/customFields_editOrderAjax.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+$_POST['address'] = '/modules/System Admin/customFields.php';
+
+require_once '../../gibbon.php';
+
+if (isActionAccessible($guid, $connection2, '/modules/System Admin/customFields.php') == false) {
+    exit;
+} else {
+    // Proceed!
+    $data = $_POST['data'] ?? [];
+    $order = json_decode($_POST['order']);
```
Comment (rossdotparker) [inline modules/System Admin/customFields_editOrderAjax.php:31]: No null coalesce here?

Comment (SKuipers) [reply modules/System Admin/customFields_editOrderAjax.php:31]: Copypasteitis. Will fix!

**Comment by rossdotparker** [comment]
Not a massive PR for a very, very impactful feature set. I've let one comment which can be dealt with post-merge if you have a moment. Going to test now.

**Pull request merged**